akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -670,9 +670,12 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
         flushingPromise.future
       }
     implicit val ec = remoteDispatcher
-    flushing.recover { case _ ⇒ Done }.map { _ ⇒
-      killSwitch.shutdown()
 
+    for {
+      _ ← flushing.recover { case _ ⇒ Done }
+      _ = killSwitch.shutdown()
+      _ ← streamsCompleted
```

Comment:
what happens if the stream futures are completed with failure for some reason?
either we just use an `onComplete` callback instead, or we have to `recover` each one?